code/modules/orbit/orbit.dm

```diff
@@ -59,7 +61,8 @@
 	lastloc = orbiter.loc
 	for(var/other_orbit in orbiter.orbiters)
 		var/datum/orbit/OO = other_orbit
-		if(OO == src || OO.orbiter == orbiting)
+		//Skip if checked already
+		if(OO in checked_already)

```

Comment:
we can cut these checks down because the initial case of an orbiter orbiting itself or another orbiter is checked by this mechanism as well